tahoe.mk

```
@@ -40,16 +32,65 @@ tahoe.restart:  ## Restarts both of LMS and Studio python processes while keepin
 
 amc.provision:  ## Initializes the AMC
 	docker exec -it tahoe.devstack.amc python manage.py migrate
-	make COMMAND='python manage.py lms create_oauth2_client http://localhost:9000/     http://localhost:9000/oauth2/access_token/ confidential --client_name AMC --client_id 6f2b93d5c02560c3f93f     --client_secret 2c6c9ac52dd19d7255dd569fb7eedbe0ebdab2db --trusted --settings=devstack_docker' SERVICE='lms' tahoe.exec.single
 	docker exec -it tahoe.devstack.amc python manage.py create_devstack_superuser
-	docker exec -it tahoe.devstack.amc-frontend npm install
+	cd $(DEVSTACK_WORKSPACE)/amc/frontend && npm install
 
 amc-shell:
 	docker exec -it tahoe.devstack.amc bash
 
 amc-frontend-shell:
-	docker exec -it tahoe.devstack.amc-frontend bash
+	@echo "AMC frontend is running in your local machine"
 
 amc-restart:
 	docker exec -t tahoe.devstack.amc bash -c 'killall5'
-	docker exec -t tahoe.devstack.amc-frontend bash -c 'killall5'
+	@echo "Note: AMC frontend is running in your local machine"
+
+amc.start.frontend:
+	@echo "Running AMC frontend  in your local machine"
+	cd $(DEVSTACK_WORKSPACE)/amc/frontend && npm start
+
+tahoe.macosx.install-nodejs-v10:
+	brew install node@10
+	brew link --overwrite --force node@10
+
+amc.check-node-version:
+	@node --version $(DEVNULL)
+ifneq ($(shell node --version | grep -o '^v10\.'), v10.)
+	echo "AMC needs Node.js v10, no more no less."
+	echo "You DO NOT have Node.js v10 installed. Please install it."
+	echo "Try running $ tahoe.macosx.install-nodejs-v10 on MacOSX."
+	exit 1
+endif
+
+tahoe.clean-provision:
+	make amc.check-node-version
+	make down
+	make tahoe.chown
+	docker volume prune --force
+
+	@# Ignoring AMC since we're building it locally.
+	docker-compose pull lms studio devpi memcached mongo mysql postgres redis
+
+	cd $(DEVSTACK_WORKSPACE) && \
+		rm -rf amc && \
+		rm -rf course-discovery && \
+		rm -rf credentials && \
+		rm -rf cs_comments_service && \
+		rm -rf ecommerce && \
+		rm -rf edx-analytics-pipeline && \
+		rm -rf edx-e2e-tests && \
+		rm -rf edx-notes-api && \
+		rm -rf edx-platform && \
+		rm -rf edx-theme-codebase && \
+		rm -rf edx-theme-customers && \
+		rm -rf src && \
+		rm -rf xqueue
+
+	make dev.clone
+
+	cd $(DEVSTACK_WORKSPACE)/amc && \
+		docker build -f Dockerfile . -t gcr.io/appsembler-tahoe-0/amc
+
+	make dev.provision
+	make amc.start.frontend
```

```suggestion
	@echo "Provisioning Tahoe is now complete. Please run the following command to start the AMC frontend (outside the Docker devstack: '$ make amc.start.frontend')"
```

@ahmedaljazzar this is what you're looking to change.

---

> Well, I don't really think freezing while provisioning is a good idea. That was totally counter intuitive. I also tried stopping the server (`make stop`) and running it again (`make tahoe.up`) but I got the a blank site on AMC with the following errors:
> 
> ```
> [Error] Failed to load resource: Could not connect to the server. (vendor.9937281d50359a7b6767.js, line 0) http://0.0.0.0:13000/static/webpack/vendor.9937281d50359a7b6767.js
> [Error] Failed to load resource: Could not connect to the server. (app.9937281d50359a7b6767.js, line 0) http://0.0.0.0:13000/static/webpack/app.9937281d50359a7b6767.js
> [Error] Failed to load resource: the server responded with a status of 404 (Not Found) (None.js, line 0) http://js.hs-scripts.com/None.js
> ```

@ahmedaljazzar I think there's a misunderstanding here, so I would like to clarify few things:

 - The whole point of this pull request is to move out the AMC Frontend outside the docker devstack.
 - Moving the AMC frontend outside the devstack means that it won't be included in the `$ docker-compose up` (or `$ make tahoe.up`), so a seperate shell running `$ make amc.start.frontend` is needed otherwise you'd get the 404 errors
 - I decided to add this as an extra step for my own convenient so I don't have to maintain two shells when I reset the devstack. I have removed this step (see https://github.com/appsembler/devstack/pull/30#discussion_r305830927), but you still have to run it on a separate terminal otherwise you'd get the 404 errors
- This is still a huge `(WIP)` PR and the `$ make amc.start.frontend` is already known to folks like Maxi and Matej so I skipped on editing the documentation till we test the PR and consider it :heavy_check_mark: